src/main/java/com/github/mmolimar/kafka/connect/fs/policy/AbstractPolicy.java

```diff
@@ -36,6 +37,8 @@
     private final FsSourceTaskConfig conf;
     private final AtomicLong executions;
     private final boolean recursive;
+    private final long batchSize;
+    private Iterator<FileMetadata> currentIterator;
```

Comment:
I'd rename this variable as ``previous`` and initialized in the constructor as ``Collections.emptyIterator()``

Reply:
Done 👍